Civi/Api4/Service/Spec/Provider/DAOFieldsCallbackAdapterSpecProvider.php

```diff
@@ -90,7 +90,10 @@ private static function legacyArrayToField(array $data, string $entityName): Fie
     $field->setType('Field');
     $field->setColumnName($name);
     $field->setNullable(empty($data['required']));
-    $field->setRequired(!empty($data['required']) && !$hasDefault && $name !== 'id');
+    // Api4 only expects field to be 'required' if the action is create.
+    if ($spec->getAction() === 'create') {
+      $field->setRequired(!empty($data['required']) && !$hasDefault && $name !== 'id');
```

Comment:
@colemanw would this impact on things like if a custom entity had a `is_test` type field being auto added to wheres?

Reply:
@seamuslee001 I don't think so, no. This simply mirrors the logic already in core:
https://github.com/civicrm/civicrm-core/blob/6a5544ab5f1b1684ca72594867d1f3e82016e3c0/Civi/Api4/Service/Spec/SpecGatherer.php#L134-L139